lib/fluent/plugin/out_logstash.rb

```diff
@@ -94,29 +100,28 @@ def write(chunk)
 
   def send_to_logstash(data)   
     retries = 0
-
     begin
-      log.trace "New attempt to logstash attempt=#{retries}" if retries > 0
-      log.trace "Send nb_event=#{data.size} events to logstash"
-      data.each do |event|
-        client.write(event)
-        @write_counter += 1
-      end
-
-    # Handle some failures
-    rescue Errno::EHOSTUNREACH, Errno::ECONNREFUSED, Errno::ETIMEDOUT, Errno::EPIPE => e
-
-      if retries < @max_retries || @max_retries == -1
-        @_socket = nil
-        a_couple_of_seconds = retries ** 2
-        a_couple_of_seconds = 30 unless a_couple_of_seconds < 30
-        retries += 1
-        log.warn "Could not push logs to logstash, attempt=#{retries} max_attempts=#{max_retries} wait=#{a_couple_of_seconds}s error=#{e.message}"
-        sleep a_couple_of_seconds
-        retry
-      end
-      raise ConnectionFailure, "Could not push logs to logstash after #{retries} retries, #{e.message}"
+      @mutex.synchronize {
```

Comment:
Since `client` is only ever called from `send_to_logstash`, perhaps the double mutex is unnecessary?